src/libraries/Common/src/System/Obsoletions.cs

```diff
@@ -43,6 +43,9 @@ internal static class Obsoletions
         internal const string CodeBaseMessage = "Assembly.CodeBase and Assembly.EscapedCodeBase are only included for .NET Framework compatibility. Use Assembly.Location instead.";
         internal const string CodeBaseDiagId = "SYSLIB0012";
 
+        internal const string EscapeUriStringMessage = "Uri.EscapeUriString can corrupt the Uri string in some cases. Consider using Uri.EscapeDataString instead.";
```

Comment:
Since `EscapeDataString` is not really a drop-in replacement of `EscapeUriString`, i.e. it should be applied to only the query string part, would it make sense to call it out explicitly, e.g.:
```suggestion
        internal const string EscapeUriStringMessage = "Uri.EscapeUriString can corrupt the Uri string in some cases. Consider using Uri.EscapeDataString only for the query string component instead.";
```